pkg/apis/core/validation/validation.go

```diff
@@ -144,6 +144,12 @@ func ValidatePodSpecificAnnotations(annotations map[string]string, spec *core.Po
 		allErrs = append(allErrs, ValidateTolerationsInPodAnnotations(annotations, fldPath)...)
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDeletionCost) {
+		if _, err := helper.GetDeletionCostFromPodAnnotations(annotations); err != nil {
```

Comment:
is ValidatePodSpecificAnnotations used to validate update requests of pods and objects containing podspecs? If so, changing validation to add new restrictions can prevent update of existing objects and keep them from being deleted successfully.

Reply:
yes

Reply:
Sorry, hit the comment button too soon :)

what do you suggest? should we drop validation completely? I am not sure about validating on create only since this annotation is supposed to be updated after creation.

Reply:
cc @wojtek-t since he suggested we validate.

Reply:
@liggitt - would the following work:
- add validation on Create()
- on Update, validate if it was previously a valid one?

Or is it still backward-incompatible?

Reply:
done.